pkg/history/repository.go

```diff
@@ -12,5 +12,5 @@ type Repository interface {
 
 	Get(ctx context.Context, id string) (RecordEntity, error)
 
-	Find(ctx context.Context, q dal.Query) ([]RecordEntity, error)
+	Find(ctx context.Context, q dal.Query) (QueryResult, error)
```

Comment:
`QueryResult` is a large structure, so has the meaning return (and takes) by pointers, not by copy?

Reply:
There are scenarios when a cursor is not provided, e.g. initial request.